homeassistant/util/yaml/loader.py

```diff
@@ -193,10 +193,10 @@ def _parse_yaml(
     """Load a YAML file."""
     # If configuration file is empty YAML returns None
     # We convert that to an empty dict
-    return (
-        yaml.load(content, Loader=lambda stream: loader(stream, secrets))  # type: ignore[arg-type]
-        or NodeDictClass()
-    )
+    result = yaml.load(content, Loader=lambda stream: loader(stream, secrets))  # type: ignore[arg-type]
+    if result is None:
+        return NodeDictClass()
```

Comment:
This seems like quite the hack, shouldn't the consumer handle this? We could also make our own extensions, i.e. `!input` and friends, do this conversion and document it.

Reply:
This is kinda odd behavior in general, why would we map an empty files to an empty dict in the first place?

Reply:
I think it's to make `!include` and friends more user friendly by treating an empty yaml file as an empty dictionary.

I think we should handle this in `config.py` though, not in our yaml loader.

Let's make this PR a draft until that's been clarified.

Reply:
We did this indeed to ensure that the top-level data structure returend from a yaml.load is always a dictionary. Let's just say that we didn't always had config validation …